dispatches/models/nuclear_case/unit_models/hydrogen_turbine_unit.py

```diff
@@ -0,0 +1,159 @@
+    CONFIG.declare("property_package_args", ConfigBlock(
+        implicit=True,
+        description="Arguments to use for constructing property packages",
+        doc="""A ConfigBlock with arguments to be passed to a property block(s)
+and used when constructing these,
+**default** - None.
+**Valid values:** {
+see property package for documentation.}"""))
+    CONFIG.declare("reaction_package", ConfigValue(
+        default=None,
+        domain=is_reaction_parameter_block,
+        description="Reaction package to use for control volume",
+        doc="""Reaction parameter object used to define reaction calculations,
+**default** - None.
+**Valid values:** {
+**None** - no reaction package,
+**ReactionParameterBlock** - a ReactionParameterBlock object.}"""))
+    CONFIG.declare("reaction_package_args", ConfigBlock(
+        implicit=True,
+        description="Arguments to use for constructing reaction packages",
+        doc="""A ConfigBlock with arguments to be passed to a reaction block(s)
+and used when constructing these,
+**default** - None.
+**Valid values:** {
+see reaction package for documentation.}"""))
+
+    def build(self):
+
+        # Call UnitModel.build to setup dynamics
+        super(HydrogenTurbineData, self).build()
+
+        self.compressor = Compressor(
+            default={"property_package": self.config.property_package})
+
+        self.stoic_reactor = StoichiometricReactor(
+            default={"property_package": self.config.property_package,
+                     "reaction_package": self.config.reaction_package,
+                     "has_heat_of_reaction": True,
+                     "has_heat_transfer": False,
+                     "has_pressure_change": False})
+
+        self.turbine = Turbine(
+            default={"property_package": self.config.property_package})
+
+        # Declare var for reactor conversion
+        self.stoic_reactor.conversion = Var(initialize=0.75, bounds=(0, 1))
+
+        # TODO: Maybe better to write at flowsheet level?
+        self.stoic_reactor.conv_constraint = Constraint(
+            expr=self.stoic_reactor.conversion * self.stoic_reactor.inlet.
+            mole_frac_comp[0, "hydrogen"] ==
+            (self.stoic_reactor.inlet.mole_frac_comp[0, "hydrogen"] -
+             self.stoic_reactor.outlet.mole_frac_comp[0, "hydrogen"]))
+
+        # Connect arcs
+        self.comp_to_reactor = Arc(
+            source=self.compressor.outlet,
+            destination=self.stoic_reactor.inlet)
+        self.reactor_to_turbine = Arc(
+            source=self.stoic_reactor.outlet,
+            destination=self.turbine.inlet)
+
+        TransformationFactory("network.expand_arcs").apply_to(self)
+
+    def initialize(self, state_args=None,
+                   solver=None, outlvl=idaeslog.NOTSET):
+
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="unit")
+
+        if solver is None:
+            init_log.warning("Solver not provided. Default solver(ipopt) "
```

Comment:
I think get `get_solver` logs a message for this as well, and also doesn't hard code the default option name.